web/react/components/sidebar.jsx

```diff
@@ -235,103 +247,143 @@ module.exports = React.createClass({
             }
         }
     },
+    onScroll: function(e) {
+        this.updateUnreadIndicators();
+    },
+    onResize: function(e) {
+        this.updateUnreadIndicators();
+    },
+    updateUnreadIndicators: function() {
+        var container = $(this.refs.container.getDOMNode());
+
+        if (this.firstUnreadChannel) {
```

Comment:
If the unread channels get updated so there is no first unread channel (perhaps from another device) How will the "$(this.refs.topUnreadIndicator.getDOMNode()).css('display', 'initial');" stuff get reset to not displaying?